### PR TITLE
Add extended data analysis

### PR DESCRIPTION
**Pull request by jonathan-paugh**
Adds functionality to perform more in-depth analysis a variety of models:
- Feature correlation
- Lambda & significant feature analysis

Adds a submenu for data selection to be used with data analysis menu options.
Adds a startup menu to select a dataset to use:
- Train dataset
- Standardized train dataset
- Test dataset

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -1,63 +1,90 @@
...
+    perform_lasso_lambda_analysis, perform_ridge_lambda_analysis, perform_feature_correlation_analysis
 from library.option_input import OptionInput
 
-def run_menu(options):
+def menu_delay():
+    """
+    Flushes terminal and waits a bit.
+    """
+    sys.stdout.flush()
+    time.sleep(0.5)
```
Comment (semibran) [inline main.py]: whats the use case for this? Might help to comment something

File: main.py [hunk 2]
```diff
@@ -1,63 +1,90 @@
...
+
+from core.loader import load_train_dataset, load_standardized_train_dataset, load_test_dataset
 from core.processing import expand_dataset, split_training_test, split_claims_accept_reject, separate_features_label
-from core.data_visualization import handle_basic_plots, handle_compound_plots
-from core.constants import DATASET_LABEL_NAME, DATASET_TRAIN_RATIO, POLYNOMIAL_ANALYSIS_DEGREES, CROSS_VALIDATION_SETS
-from core.data_analysis import calculate_linear_regression_error, calculate_polynomial_complexity_errors
+from core.data_visualization import generate_data_visualization_plots
+from core.constants import DATASET_LABEL_NAME, DATASET_TRAIN_RATIO, MENU_EXIT, MENU_RETURN
+from core.data_analysis import perform_linear_regression_analysis, perform_polynomial_complexity_analysis, \
+    perform_lasso_lambda_analysis, perform_ridge_lambda_analysis, perform_feature_correlation_analysis
```
Comment (semibran) [inline main.py:9]: fwiw i am separating out the decision tree part into its own `main_induction.py` that can be tested separately from the main driver and also exposes the perform component for use inside the main menu.. I think that would be more scalable than cramming everything into data_analysis but its your call

**Review by semibran — APPROVED** [review]

File: main.py (hunk 2 above)
Comment (jonathan-paugh) [reply main.py:9]: That sounds good to me.

File: main.py (hunk 1 above)
Comment (jonathan-paugh) [reply main.py]: This was for temporary testing, will remove it.

**Pull request merged**